python/lsst/ts/standardscripts/maintel/set_dof.py

```diff
@@ -42,11 +47,131 @@ class SetDOF(ApplyDOF):
+        schema_dict = yaml.safe_load(schema_yaml)
+
+        base_schema_dict = super(ApplyDOF, cls).get_schema()
+
+        for prop in base_schema_dict["properties"]:
+            schema_dict["properties"][prop] = base_schema_dict["properties"][prop]
+
+        return schema_dict
+
+    async def configure(self, config) -> None:
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        super(ApplyDOF, self).configure(config)
+
+        if hasattr(config, "day") and hasattr(config, "seq"):
+            self.day = config.day
+            self.seq = config.seq
+
+    async def get_image_time(self, day, seq):
+        """Get the image time from the given day and sequence number.
+
+        Parameters
+        ----------
+        day : `int`
+            Day obs to be used for synchronizing the state.
+        seq : `int`
+            Sequence number to be used for synchronizing the state.
+
+        Returns
+        -------
+        image_time : `datetime.datetime`
+            Image time.
+        """
+
+        query = f"""
+            SELECT  "imageDate", "imageNumber"
+            FROM "efd"."autogen"."lsst.sal.CCCamera.logevent_endReadout"
+            WHERE imageDate =~ /{day}/ AND imageNumber = {seq}
+        """
+
+        time = await self.client.influx_client.query(query)
+        return Time(time.iloc[0].name)
+
+    async def get_last_issued_state(self, time):
+        """Get the state from the given day and sequence number.
+
+        Parameters
+        ----------
+        time : `datetime.datetime`
+            Initial time to query.
+
+        Returns
+        -------
+        state : `dict`
+            State of the system.
+        """
+
+        topics = [f"aggregatedDoF{i}" for i in range(50)]
+
+        current_end = time
+        lookback_interval = TimeDelta(1, format="jd")
+        max_lookback_days = 7
+
+        while True:
+            current_start = current_end - lookback_interval
+
+            state = await self.client.select_time_series(
+                "lsst.sal.MTAOS.logevent_degreeOfFreedom",
+                topics,
+                current_start,
+                current_end,
+            )
+
+            if not state.empty:
+                state = state.iloc[[-1]]
+                return state.values.squeeze()
+
+            current_end = current_start
+            if (time - current_end).jd > max_lookback_days:
+                self.log.info(
+                    "No data found within the specified range and maximum lookback period."
+                )
+                return np.zeros(50)
+        return state.values.squeeze()
+
     async def run(self) -> None:
         """Run script."""
         # Assert feasibility
         await self.assert_feasibility()
 
+        if self.day is not None and self.seq is not None:
+            self.client = EfdClient("summit_efd")
```

Comment:
note that you are hard-coding which EFD to use here, which means we won't be able to test this script at BTS or TTS. 

Consider doing something like the `BaseBlockScript` script does. See [here](https://github.com/lsst-ts/ts_standardscripts/blob/cf97e29d73c40665d1aeecfc806a5038bfcec12e/python/lsst/ts/standardscripts/base_block_script.py#L38-L42) and [here](https://github.com/lsst-ts/ts_standardscripts/blob/cf97e29d73c40665d1aeecfc806a5038bfcec12e/python/lsst/ts/standardscripts/base_block_script.py#L162-L177).